source/winVersion.py

```diff
@@ -26,35 +27,38 @@ def __init__(
 			major: int = 0,
 			minor: int = 0,
 			build: int = 0,
+			releaseName: str = "",
 			servicePack: str = "",
 			productType: str = ""
 	):
 		self.major = major
 		self.minor = minor
 		self.build = build
+		self.releaseName = releaseName
 		self.servicePack = servicePack
 		self.productType = productType
 
 	def _windowsVersionToReleaseName(self):
-		"""Returns release names for a given Windows version.
+		"""Returns release names for a given Windows version if not defined.
 		For example, 6.1 will return 'Windows 7'.
 		For Windows 10, feature update release name will be included.
-		On server systems, client release names will be returned.
+		On server systems, unless noted otherwise, client release names will be returned.
 		For example, 'Windows 10 1809' will be returned on Server 2019 systems.
 		"""
+		if self.releaseName:
+			return self.releaseName
 		if (self.major, self.minor) == (6, 1):
 			return "Windows 7"
 		elif (self.major, self.minor) == (6, 2):
 			return "Windows 8"
 		elif (self.major, self.minor) == (6, 3):
 			return "Windows 8.1"
 		elif self.major == 10:
-			buildsToReleases = {build: release for release, build in WIN10_RELEASE_NAME_TO_BUILDS.items()}
-			if self.build in buildsToReleases:
-				return f"Windows 10 {buildsToReleases[self.build]}"
-			else:
-				# Windows Insider build.
-				return "Windows 10 prerelease"
+			# From Version 1511 (build 10586), release Id/display version comes from Windows Registry.
+			# Always return "Windows 10 1507" on build 10240.
+			if self.build == 10240:
+				return "Windows 10 1507"
+			return f"Windows 10 {WIN10CURRENTRELEASENAME}"

```

Comment:
I don't think this should fall back to the current release name, that would be quite confusing.

Reply:
If we want to support manual construction of these classes rather than only using the pre-constructed ones `WIN10_1607` etc, and automatically supplying the release name where it is known, then instead of passing in the name on construction make a build to release name map and look up the release name from the build.